### PR TITLE
Feature/necroplant polishment

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- New  `Necroplant` model.
- New  `Necroplant` VFX.
- New  `Necroplant` sounds.
- `Necroplant` can spawn.
- New mechanic where `Necroplant` digs away from Player.

### CODE REVIEW

File: Scripts/src/Script/EnemyController.cpp [hunk 1]
```diff
@@ -451,6 +451,69 @@ void EnemyController::InitVFX()
...
+	{
+		hit_particles_VFX->Stop();
+	}
+
+	if (death_effect != nullptr)
+	{
+		death_effect->Disable();
+	}
+}
+
```
Comment (Dartemiss) [inline Scripts/src/Script/EnemyController.cpp:516]: This probably can be const

File: Scripts/src/Script/NecroExplosion.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+	explosion_source->PlayEvent("Play_Necroprojectile_Hit");
+}
+
+void NecroExplosion::Hide()
+{
+	explosion_mesh->SetEnabled(false);
+	explosion_particles->Stop();
+	explosion_source->StopAll();
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/NecroExplosion.cpp:108]: This can be const

File: Scripts/src/Script/NecroMuzzle.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	muzzle_mesh_renderer->material_to_render->diffuse_color[1] = muzzle_color.y;
+	muzzle_mesh_renderer->material_to_render->diffuse_color[2] = muzzle_color.z;
+
+	shoot_particles->Emit(10);
+}
+
+void NecroMuzzle::Hide()
+{
+	muzzle_mesh->SetEnabled(false);
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/NecroMuzzle.cpp:96]: Can be const

File: Scripts/src/Script/NecroProjectile.cpp [hunk 4]
```diff
@@ -85,19 +113,24 @@ bool NecroProjectile::IsDead() const
...
+
 	is_alive = true;
 	alive_time = 0.f;
+
 	projectile_mesh_go->SetEnabled(true);
+	projectile_beam_go->SetEnabled(true);
 }
 
-void NecroProjectile::Kill()
+void NecroProjectile::Hide()
```
Comment (Dartemiss) [inline Scripts/src/Script/NecroProjectile.cpp:126]: Const

**Review by Dartemiss — APPROVED** [review]
Very good job Mario! Is amazing, i love it! The only thing I would say is that is very strong I couldn't pass the third battle zone xD But that is balancing.

Just the check the comments on the code :D

File: Scripts/src/Script/EventManager.h [hunk 5]
```diff
@@ -70,8 +70,10 @@ class EventManager : public Script
 
 	void PlayAudioOnAwake(std::vector <GameObject*> gos_to_play);
 
-private:
+public:
+	BattleEvent* current_battle = nullptr;
```
Comment (JorxSS) [inline Scripts/src/Script/EventManager.h:74]: there is a public zone declared at the end of the file, so move this line there, or those here.

**Review by JorxSS — APPROVED** [review]
Incredible work mate! This works flawless!! GREAT JOB!

image

**Pull request merged**